For example https://cirrus-ci.com/task/5642614044098560?logs=ci#L3315

```
 node0 2022-05-13T06:33:06.355942Z [msghand] [net_processing.cpp:4178] [MaybeDiscourageAndDisconnect] Warning: disconnecting but not discouraging local peer 1! 
 node0 2022-05-13T06:33:06.355985Z [net] [net.cpp:567] [CloseSocketDisconnect] disconnecting peer=1 
 test  2022-05-13T06:33:06.356000Z TestFramework.p2p (DEBUG): Closed connection to: 127.0.0.1:11408 
 test  2022-05-13T06:33:06.356000Z TestFramework.p2p (DEBUG): Send message to 127.0.0.1:11408: msg_ping(nonce=00000007) 
 test  2022-05-13T06:33:06.356000Z TestFramework (ERROR): Unexpected exception caught during testing 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 133, in main
                                       self.run_test()
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_unrequested_blocks.py", line 264, in run_test
                                       test_node.sync_with_ping(timeout=1)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 555, in sync_with_ping
                                       self.send_message(msg_ping(nonce=self.ping_counter))
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 266, in send_message
                                       return self.send_raw_message(tmsg)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 270, in send_raw_message
                                       raise IOError('Not connected')
                                   OSError: Not connected

--- a/test/functional/p2p_unrequested_blocks.py
+++ b/test/functional/p2p_unrequested_blocks.py
@@ -257,16 +257,11 @@ class AcceptBlockTest(BitcoinTestFramework):
         test_node.send_message(msg_block(block_291))
 
         # At this point we've sent an obviously-bogus block, wait for full processing
-        # without assuming whether we will be disconnected or not
-        try:
-            # Only wait a short while so the test doesn't take forever if we do get
-            # disconnected
-            test_node.sync_with_ping(timeout=1)
-        except AssertionError:
-            test_node.wait_for_disconnect()
+        # and assume disconnection
+        test_node.wait_for_disconnect()
 
-            self.nodes[0].disconnect_p2ps()
-            test_node = self.nodes[0].add_p2p_connection(P2PInterface())
+        self.nodes[0].disconnect_p2ps()
+        test_node = self.nodes[0].add_p2p_connection(P2PInterface())
 
         # We should have failed reorg and switched back to 290 (but have block 291)
         assert_equal(self.nodes[0].getblockcount(), 290)

